pkg/addon/config.go

```diff
@@ -0,0 +1,214 @@
+package addon
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strconv"
+
+	"github.com/ViaQ/logerr/log"
+	"github.com/openshift/cluster-logging-operator/pkg/apis"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	elasticsearch "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const addonSecretName = "addon-cluster-logging-operator-parameters"
+
+var ErrNoAddonConfig = errors.New("No addon configiration found")
+
+type AddonConfiguration struct {
+	EsMaxRetention string `data-field:"es-max-retention" default:"7d"`
```

Comment:
query: what is data-field here?

Reply:
This is used to match the field name from the secret to the `AddonConfiguration` struct field